flooder/src/main.rs

```diff
@@ -0,0 +1,308 @@
+mod config;
+
+use clap::Parser;
+use codec::{Compact, Encode};
+use config::Config;
+use hdrhistogram::Histogram as HdrHistogram;
+use log::{debug, info, warn};
+use rayon::prelude::*;
+use sp_core::{sr25519, DeriveJunction, Pair};
+use sp_runtime::{generic, traits::BlakeTwo256, MultiAddress, OpaqueExtrinsic};
+use std::sync::{Arc, Mutex};
+use std::thread::sleep;
+use std::time::{Duration, Instant};
+use substrate_api_client::rpc::WsRpcClient;
+use substrate_api_client::{
+    compose_call, compose_extrinsic_offline, AccountId, Api, GenericAddress, UncheckedExtrinsicV4,
+    XtStatus,
+};
+
+type TransferTransaction =
+    UncheckedExtrinsicV4<([u8; 2], MultiAddress<AccountId, ()>, codec::Compact<u128>)>;
+type BlockNumber = u32;
+type Header = generic::Header<BlockNumber, BlakeTwo256>;
+type Block = generic::Block<Header, OpaqueExtrinsic>;
+
+fn main() -> Result<(), anyhow::Error> {
+    env_logger::init();
+    let config: Config = Config::parse();
+    info!("Starting benchmark with config {:#?}", &config);
+
+    let account = match config.phrase {
+        Some(phrase) => {
+            sr25519::Pair::from_phrase(&config::read_phrase(phrase), None)
+                .unwrap()
+                .0
+        }
+        None => match config.seed {
+            Some(seed) => sr25519::Pair::from_string(&seed, None).unwrap(),
+            None => panic!("Needs --phrase or --seed"),
+        },
+    };
+
+    let pool = create_connection_pool(config.nodes);
+    let connection = pool.get(0).unwrap();
+
+    let total_users = config.transactions;
+    let transactions_per_batch = config.throughput / config.threads;
```

Comment:
Now it should be something like 
```suggestion
    let transactions_per_batch = config.throughput / current_num_threads();
```